### PR TITLE
Updated Docker Development mode documentation

### LINKED ISSUE #73: What is "CONTAINER ID CREATED ABOVE"?
The instructions for running this using Docker reference what I assume is a placeholder stating, "(CONTAINER ID CREATED ABOVE)". From where does a user get this value? The README is unclear on this.

### PR DESCRIPTION
**Pull request by dhruv282**
Used solution provided in #73 
Fixes #73

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,47 +57,17 @@ Clone the repository and change working directory (if not already) then build th
 $ git clone https://github.com/oduwsdl/tmvis.git
 $ cd tmvis
 $ docker image build -t timemapvis .
+$ docker run --shm-size=1G -it --rm -v "$PWD":/app -v /app/node_modules -p 3000:3000 --user=$(id -u):$(id -g) timemapvis node tmvis.js
```
Comment (ibnesayeed) [inline README.md:60]:
```suggestion
$ docker run --shm-size=1G -it --rm -p 3000:3000 timemapvis
```

To run the application in a simple mode, you do not need any of that extra stuff. You might document the location of the cache/screenshots directory so people can bind mount that volume for persistence, but you do not want them to mount the `$PWD` in `/app` which is the use case of development mode.

File: README.md [hunk 2]
```diff
@@ -57,47 +57,17 @@ Clone the repository and change working directory (if not already) then build th
...
-$ git clone https://github.com/oduwsdl/tmvis.git
-$ cd tmvis
-$ docker image build -t timemapvis .
-$ docker container run --shm-size=1G -it --rm -v "$PWD":/app --user=$(id -u):$(id -g) timemapvis npm install
-$ docker container run --shm-size=1G -it --rm -v "$PWD":/app -p 3000:3000 --user=$(id -u):$(id -g) timemapvis
-
-```
-
-Once the image is built and dependencies are installed locally under the `node_modules` directory of the local clone, only the last command would be needed for continuous development. Since the default container runs under the `root` user, there might be permission related issues on the `npm install` step. If so, then try to manually create the `node_modules` directory and change its permissions to world writable (`chmod -R a+w node_modules`) then run the command to install dependencies again.
+In case if you want to make changes in the `tmvis` code itself, you might want to run it in the development mode by mounting the code from the host machine inside the container so that changes are reflected immediately, without requiring an image rebuild.
```
Comment (ibnesayeed) [inline README.md:70]: This is where you would want to mount `$PWD` and declare `node_modules` volume to allow development mode. Note that mapping UID/GID is not needed if you do not worry about files on the host being written by the root user from the container. You can add that as an additional note, if someone wants to keep files under the ownership of the host user. Another point to note here is, if you are not supplying any arguments, there is no need to specify the command to run inside i.e., `node tmvis.js`, that is already the default `CMD` in the Dockerfile. If you want top execute another script/command, then you need to explicitly supply it.

```
$ docker run --shm-size=1G -it --rm -v "$PWD":/app -v /app/node_modules -p 3000:3000 timemapvis
```

File: README.md (hunk 1 above)
Comment (dhruv282) [reply README.md:60]: This would make the code block match the non development mode block defined earlier in the file. Do you think it would be better to get rid of the non development block to avoid confusion and rename the heading of the development mode to say "Running as a Docker Container"?

Comment (ibnesayeed) [reply README.md:60]: There should be just one section for Docker in which you start from a simple use case, then talk about necessary volumes for persistent storage, and finally discuss what is needed if developers want to make their local changes reflected without rebuilding the image each time (unless a dependency changes). The whole naive and experimental annotation is unnecessary and confusing.

Comment (dhruv282) [reply README.md:60]: Gotcha, I'll get rid of that and make appropriate changes. Thanks!

**Comment by dhruv282** [comment]
I ended up restructuring the whole file and added some missing info

**Comment by ibnesayeed** [comment]
This looks much better. Though, I would suggest you test changes in the Docker related section, in case there are corner cases that we might have missed. 

I would actually suggest you begin Docker section with using our published docker image `oduwsdl/tmvis` without the need of asking them to build it locally first.

```
$ docker run --shm-size=1G -it --rm -p 3000:3000 oduwsdl/tmvis
```

Then go on to suggest if they want to build a custom image from the source, they can clone the code, build an image and run it instead (i.e., the currently listed commands).

**Comment by dhruv282** [comment]
I have tested the changes and everything is working as expected

File: README.md [hunk 3]
```diff
@@ -6,9 +6,14 @@ Archives"](http://www.cs.odu.edu/~mln/pubs/ecir-2014/ecir-2014.pdf) for the Web
 
 ## Running as a Docker Container
 
-Running the server in a [Docker](https://www.docker.com/) container can make the process of dependency management easier. The code is shipped with a `Dockerfile` to build a Docker image that will run the service when started. This document assumes that you have Docker setup already, if not then follow the [official guide](https://docs.docker.com/installation/).
+Running the server in a [Docker](https://www.docker.com/) container can make the process of dependency management easier. This document assumes that you have Docker setup already, if not then follow the [official guide](https://docs.docker.com/installation/).
 
-Clone the repository and change the working directory then build and run the image as the following:
+A published Docker image can be used as shown below:
```
Comment (ibnesayeed) [inline README.md:11]:
```suggestion
A Docker image from the current version of the code is built and published on Docker Hub at [oduwsdl/tmvis](https://hub.docker.com/r/oduwsdl/tmvis). Pull the image and run the service in a container as shown below:

```
